src/services/token/tokenOwnerHook.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ *  TODO: Extract this code out to its own package to reuse in tradetrust-website
```

Comment:
lol

Reply:
develop it here first for fast feedback but let's move all of this to a library